packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/interact/content/useEnsAddress.ts

```diff
@@ -0,0 +1,33 @@
+import { createPublicClient, http, isAddress } from "viem";
+import { getEnsAddress } from "viem/actions";
+import { mainnet } from "viem/chains";
+import { normalize } from "viem/ens";
+import { useQuery } from "@tanstack/react-query";
+
+const mainnetClient = createPublicClient({
+  chain: mainnet,
+  transport: http(),
+});
+
+function isValidEnsName(name: string): boolean {
+  return !!name && name.includes(".") && name.endsWith(".eth") && !isAddress(name);
```

Comment:
```suggestion
  return name.includes(".");
```

- ENS supports any DNS name, not just `.eth`
- if a name has a `.`, it already isn't an address, so no need to check
- if a name has a `.`, it already is truthy, so no need to check

given this, you could probably just `enabled: name.includes(".")` below

Reply:
yeah makes sense, committed the suggestion